docs/package/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+    }
+
+    # cmake generator as a bare minimum
+    # cmake_find_package if at least one dependency is listed in https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules
+    # cmake_find_package_multi for other dependencies (but cmake_find_package may be a better choice if you have to take precedence over custom Find modules files in upstream source code).
+    # pkg_config if at least one dependency is discovered with pkg_check_modules in upstream CMakeLists (don't forget to also add pkgconf in build requirements)
+    generators = "cmake", "cmake_find_package_multi"
+
+    # no manual caching of build helper like  _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    # don't use self.settings_build
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    # don't use self.user_info_build
+    @property
+    def _user_info_build(self):
+        return getattr(self, "user_info_build", self.deps_user_info)
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx # for plain C projects only
+        del self.settings.compiler.cppstd # for plain C projects only
+
+    def requirements(self):
+        self.requires("dependency/0.8.1")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration("package can't be built as shared on visual studio")
+
+    # if another tool than the compiler or CMake is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
+        self.build_requires("tool/x.y.z")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        # remove bundled xxhash
+        tools.remove_files_by_mask(os.path.join(self._source_subfolder, "lib"), "whateer.*")
+        tools.replace_in_file(
+            os.path.join(self._cmakelists_subfolder, "CMakeLists.txt"),
+            "...",
+            "",
+        )
+
+    @functools.lru_cache(1)
```

Comment:
What does that mean ?

Reply:
it's a [python thing](https://docs.python.org/3/library/functools.html#functools.lru_cache): when calling this function, python actually does the call once, and caches the resulting object. When the same function is called after that, the cached result is returned.